CONTRIBUTING.md

```diff
@@ -0,0 +1,182 @@
+# Contributing Guidelines
+
+This documentation contains a set of guidelines to help you during the contributing process. We are happy to welcome all the contributors who's willing to enhance/add features to this project. Feel free to contribute and remember, no contribution is too small.
+
+## Submitting Contributions :woman_technologist: :man_technologist:
+
+### Step 1 : Find an issue
+
+- Take a look at the Existing Issues or [create your own Issues!](#creating-an-issue)
+- Wait for the Issue to be assigned to you after which you can start working on it.
+- Note : Every change in this project should/must have an associated issue.
+
+### Step 2: Fork the repository
+
+- Fork this Repository. This will create a local copy of this repository on your Github Profile.
+
+![fork](https://user-images.githubusercontent.com/66238394/114071871-3cbb1080-98bf-11eb-94aa-4b5f811ece77.png)
+
+### Step 3: Clone the repository:
+
+- Clone this Github repo. Open up the GitBash/Command Line and type in:
+
+```git
+$ git clone https://github.com/YOUR_USERNAME/NeoAlgo-Docs.git
+```
+
+Don't forget to write your GitHub user name in place of _YOUR_USERNAME_
+
+- Keep a reference to the original project in upstream remote.
+
+```git
+$ cd NeoAlgo-Docs
+$ git remote add upstream https://github.com/TesseractCoding/NeoAlgo-Docs
+```
+
+- Update your copy before working.
+
+```git
+$ git remote update
+$ git checkout <branch-name>
+$ git rebase upstream/<branch-name>
+```
+
+### Step 4: Create a development branch
+
+- `cd` into the parent directory. Now create a branch of the master by pushing the command
+
+```git
+$ git branch YOUR_USERNAME
+```
+
+You can now check the branches by pushing in the command: git branch. You will see the Master Branch and another branch with your username.
+
+- Switch to Development Branch. Never develop on Master Branch. Push in the command:
+
+```git
+$ git checkout YOUR_USERNAME
+```
+
+Voila! You've completed the initial setup :smiley:
+
+Start Coding 👩‍💻👨‍💻
+
+### Step 5: Commit your changes
+
+- Stage the files that you want to include in your commit. To stage a specific changed file, type in:
+
+  ```git
+  $ git add <FILE_NAME>
+  ```
+
+  To stage all the changed files, type in:
+
+  ```git
+  $ git add --all
+  ```
+
+  This command snapshots the file in preparation for versioning, adding it to the staging area.
```

Comment:
` This command snapshots the file in preparation for versioning and further adds it to the staging area.`